### ISSUE TITLE
Bug: FPE_OVERFLOW in SeedFinder.ipp

### ISSUE THREAD

**Issue by pbutti** [acts-project/acts#2046]
I think we found a reproducible full_chain_odd.py crash at event 322 when running Fatras with ttbar due to PE_OVERFLOW
The crash seems to be caused by  
https://github.com/acts-project/acts/blob/61c067e161839f429353b7ee79826562dbb95e2e/Core/include/Acts/Seeding/SeedFinder.ipp#L648
(@benjaminhuth nailed it down)
Can be reproduced with current acts main branch.

**Comment by benjaminhuth** [comment]
Tagging @LuisFelipeCoelho for this

**Comment by andiwand** [comment]
could you take a look at this @LuisFelipeCoelho ?

**Comment by paulgessinger** [comment]
Is there any chance you could build with debug symbols (`RelWithDebInfo`) and ideally run this throug gdb (like `gdb --args python ./Examples/Scripts/Python/full_chain_odd.py --ttbar` to get resolved symbols, so we can try to figure out where exactly it's throwing the FPE?

**Comment by owtscharenko** [comment]
Yes of course. 
After rebuilding with `-DCMAKE_BUILD_TYPE=RelWithDebInfo` and executing 
`gdb python`
and then
`run ./Examples/Scripts/Python/full_chain_odd.py --ttbar`
I get the following

```
15:16:20    Sequencer      INFO      finished event 50
15:16:26    AmbiguityRes   INFO      Resolved to 198 tracks from 1943
15:16:26    IterativeVer   INFO      Found 14 vertices in event
15:16:26    IterativeVer   INFO      Found vertex at -0.0146702 -0.0158569    28.3381          0 with 21 tracks.
15:16:26    IterativeVer   INFO      Found vertex at  0.0114731 -0.0114536   -17.0755          0 with 28 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.0198567 0.00759836    48.7533          0 with 20 tracks.
15:16:26    IterativeVer   INFO      Found vertex at   0.0130136 -0.00305421     36.9669           0 with 16 tracks.
15:16:26    IterativeVer   INFO      Found vertex at 0.00318279 -0.0114043    -26.518          0 with 15 tracks.
15:16:26    IterativeVer   INFO      Found vertex at  -0.0475335 0.000745443    -46.7306           0 with 8 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.0123693  0.0143468    63.5447          0 with 16 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.00403448  -0.0222971    -53.3036           0 with 8 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.0279412 -0.0181979   -79.2599          0 with 9 tracks.
15:16:26    IterativeVer   INFO      Found vertex at 0.0262615 0.0078358   11.7178         0 with 12 tracks.
15:16:26    IterativeVer   INFO      Found vertex at 0.0172055 0.0318612   30.8791         0 with 6 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.197278 0.0631813   129.664         0 with 4 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.0570241   0.022466    74.8053          0 with 4 tracks.
15:16:26    IterativeVer   INFO      Found vertex at -0.0135896  -0.012535   -62.5757          0 with 3 tracks.
15:16:26    VertexPerfor   INFO      Total number of reco track-associated truth particles in event : 198
15:16:26    VertexPerfor   INFO      Total number of reco track-associated truth primary vertices : 21
15:16:26    Sequencer      INFO      finished event 51

Program received signal SIGFPE, Arithmetic exception.
0x00007ffff6ba5e6a in Acts::SeedFinder<ActsExamples::SimSpacePoint, void*>::filterCandidates (this=0x555565f91bb0, spacePointData=..., spM=..., options=..., seedFilterState=..., state=...) at /home/niko/git/acts/Core/include/Acts/Seeding/SeedFinder.ipp:698
698             float pT = options.pTPerHelixRadius * std::sqrt(S2 / B2) / 2.;
(gdb) 
```

[gdb_output.txt](https://github.com/acts-project/acts/files/11373511/gdb_output.txt)

**Comment by owtscharenko** [comment]
I looked into the given line of code, and asked gdb to print the variables used for calculation of pT :
```
(gdb) p S2
$3 = 1.01071525
(gdb) p B2
$4 = 0
```
and B2 is 0, explaining the `SIGFPE`. 
I looked into line [678](https://github.com/acts-project/acts/blob/cb8d0b03d962b06ec31cb9ac9eb67861ec3d43f2/Core/include/Acts/Seeding/SeedFinder.ipp#L678) where `B = Vb - A * Ub`  is calculated, and from the variables I calculate B to 5.19e-11 and `B2 = B* B` = 2.69e-21.
```
(gdb) p Vb
$19 = 0.00299345562
(gdb) p A
$20 = -0.103514351
(gdb) p Ub
$21 = -0.0289182663
```
I am not familiar with the calculation of the helix in u-v space and can not really judge what this means. I guess in case of B2==0 one could default to a `pTscatterSigma` calculated from `maxPtScattering` ?
